### PR TITLE
SetDetScale option for text file input

### LINKED ISSUE #17946: SetDetScale should have text input file
String input is difficult for instruments with many detectors

### PR DESCRIPTION
**Pull request by VickieLynch**
Description of work.
File input was requested since text string is too long for many detectors.
**To test:**
```python
Load(Filename='TOPAZ_3007.peaks', OutputWorkspace='TOPAZ_3007')
SetDetScale(Workspace='TOPAZ_3007', DetScaleFile='setDetScale.txt')

wksp = mtd['TOPAZ_3007']
print wksp.getInstrument().getNumberParameter("detScale49")
```

Fixes #17946 
[setDetScale.txt](https://github.com/mantidproject/mantid/files/592526/setDetScale.txt)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/SetDetScale.py [hunk 1]
```diff
@@ -51,6 +56,14 @@ def PyExec(self):
             comp, value = component.split(":")
```
Comment (gvardany) [inline Framework/PythonInterface/plugins/algorithms/SetDetScale.py]: I would expect that if both the list and the file are given, the list should overwrite the scales given in the file. 
So I would move this loop after the loop over the lines in the file.

**Review by gvardany — CHANGES_REQUESTED** [review]
I had one comment (see inline). Also could you please add a release note and update the documentation of the algorithm?

**Comment by gvardany** [comment]
It seems like overwriting the scale does not work, even after loops are swapped. 
Did you manage to test this case?
Maybe one needs to organise the list better, to make sure that the key (i.e. detector ID) is unique in the list?
Anyway, would be nice to have one more unit test for this corner case.

**Comment by VickieLynch** [comment]
I am using a dictionary now and have added test.  See if overwriting works for you.

Vickie E Lynch
S&T Staff
Neutron Data Analysis & Visualization Division

Oak Ridge National Laboratory
PO Box 2008, Oak Ridge, TN 37831-6475
Office: 865-250-4795 | Mobile: 865-250-4795 | Fax: 865-574-6080
lynchve@ornl.gov | Bldg. 8600, Room C-448, MS 6475

**Review by gvardany — APPROVED** [review]

**Comment by gvardany** [comment]
Sorry it was my bad, I did not pull your previous commit, but you still caught the email.
Anyway I agree that dictionary here is way more appropriate, and it works nice.
:shipit:
